### PR TITLE
implement unpairdevice API for python CHIP controller

### PR DESCRIPTION
**Pull request by genesiscrew**
generic CHIP controller library recently implemented the unpairDevice API in pull request #24127, but the python CHIP controller which is based on the generic library did not, so I added functionality to python CHIP controller.

### CODE REVIEW

File: examples/lighting-app/esp32/sdkconfig.optimize.defaults [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#
```
Comment (bzbarsky-apple) [inline examples/lighting-app/esp32/sdkconfig.optimize.defaults:1]: What does this have to do with the Python controller?

File: scripts/build/build/targets.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def BuildHostFakeTarget():
     target.AppendModifier("asan", use_asan=True).ExceptIfRe("-tsan")
     target.AppendModifier("tsan", use_tsan=True).ExceptIfRe("-asan")
     target.AppendModifier("ubsan", use_ubsan=True)
-    target.AppendModifier("libfuzzer", use_tsan=True).OnlyIfRe("-clang")
+    target.AppendModifier("libfuzzer", use_libfuzzer=True).OnlyIfRe("-clang")
```
Comment (bzbarsky-apple) [inline scripts/build/build/targets.py:71]: This seems unrelated to the Python controller.

File: src/controller/python/ChipDeviceController-ScriptBinding.cpp [hunk 3]
```diff
@@ -386,6 +390,41 @@ PyChipError pychip_DeviceController_ConnectWithCode(chip::Controller::DeviceComm
...
+{
+    // Create a new CurrentFabricRemover instance
+    auto * fabricRemover = new chip::Controller::CurrentFabricRemover(devCtrl);
+
+    auto * callbacks = new UnpairDeviceCallback(callback, fabricRemover);
+
+    // Pass the callback and nodeid to the RemoveCurrentFabric function
+    CHIP_ERROR err = fabricRemover->RemoveCurrentFabric(nodeid, &callbacks->mOnCurrentFabricRemove);
+
+    return ToPyChipError(err);
```
Comment (bzbarsky-apple) [inline src/controller/python/ChipDeviceController-ScriptBinding.cpp:425]: On error return from `RemoveCurrentFabric`, doesn't this leak?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/controller/python/ChipDeviceController-ScriptBinding.cpp (hunk 3 above)
Comment (genesiscrew) [reply src/controller/python/ChipDeviceController-ScriptBinding.cpp]: fixed.

File: examples/lighting-app/esp32/sdkconfig.optimize.defaults (hunk 1 above)
Comment (genesiscrew) [reply examples/lighting-app/esp32/sdkconfig.optimize.defaults:1]: yes its unrelated, mistakenly pulled latest changes from master  via clicking update branch button on github..

File: scripts/build/build/targets.py (hunk 2 above)
Comment (genesiscrew) [reply scripts/build/build/targets.py:71]: yes its unrelated, mistakenly pulled latest changes from master  via clicking update branch button on github..

File: src/controller/python/ChipDeviceController-ScriptBinding.cpp (hunk 3 above)
Comment (bzbarsky-apple) [reply src/controller/python/ChipDeviceController-ScriptBinding.cpp]: This is not fixed.  Just ignoring the error does not fix the leak....

Comment (genesiscrew) [reply src/controller/python/ChipDeviceController-ScriptBinding.cpp:429]: yeap your right, I fixed now hopefully..

File: src/controller/python/ChipDeviceController-ScriptBinding.cpp [hunk 4]
```diff
@@ -386,6 +390,45 @@ PyChipError pychip_DeviceController_ConnectWithCode(chip::Controller::DeviceComm
...
+    // Create a new CurrentFabricRemover instance
+    auto * fabricRemover = new chip::Controller::CurrentFabricRemover(devCtrl);
+
+    auto * callbacks = new UnpairDeviceCallback(callback, fabricRemover);
+
+    // Pass the callback and nodeid to the RemoveCurrentFabric function
+    CHIP_ERROR err = fabricRemover->RemoveCurrentFabric(nodeid, &callbacks->mOnCurrentFabricRemove);
+    if (err != CHIP_NO_ERROR)
+    {
+        delete fabricRemover;
```
Comment (bzbarsky-apple) [inline src/controller/python/ChipDeviceController-ScriptBinding.cpp:426]: What about `callbacks`?

Comment (genesiscrew) [reply src/controller/python/ChipDeviceController-ScriptBinding.cpp:426]: correct, is cleaned now as well.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/lighting-app/esp32/sdkconfig.optimize.defaults (hunk 1 above)
Comment (andy31415) [reply examples/lighting-app/esp32/sdkconfig.optimize.defaults:1]: This file still shows up.

**Review by andy31415 — APPROVED** [review]
